modules/liquidity-pools/src/lib.rs

```diff
@@ -158,13 +169,84 @@ impl<T: Trait> Module<T> {
 	}
 
 	pub fn is_enabled(pool_id: T::LiquidityPoolId, currency_id: T::CurrencyId, leverage: Leverage) -> bool {
-		match <LiquidityPoolOptions<T>>::get(&pool_id, &currency_id) {
+		match Self::liquidity_pool_options(&pool_id, &currency_id) {
 			Some(pool) => pool.enabled_longs.contains(leverage) || pool.enabled_shorts.contains(leverage),
 			None => false,
 		}
 	}
 }
 
+impl<T: Trait> LiquidityPoolBaseTypes for Module<T> {
+	type LiquidityPoolId = T::LiquidityPoolId;
+	type CurrencyId = T::CurrencyId;
+}
+
+impl<T: Trait> LiquidityPoolsConfig<T::AccountId> for Module<T> {
+	fn get_bid_spread(pool_id: Self::LiquidityPoolId, currency_id: Self::CurrencyId) -> Option<Permill> {
+		match Self::liquidity_pool_options(&pool_id, &currency_id) {
```

Comment:
`Self::liquidity_pool_options(&pool_id, &currency_id).map(|pool| pool.bid_spread)`

and few others below